src/trap.h

```diff
@@ -61,79 +57,132 @@ typedef void (trapfunc::*trap_function)( Creature *, int x, int y );
         }
+        /**
+         * Whether triggering the trap can be avoid (if greater than 0) and if so, this is
+         * compared to dodge skill (with some adjustments). Smaller values means it's easier
+         * to dodge.
+         */
         int get_avoidance() const
         {
             return avoidance;
         }
+        /**
+         * This is used when disarming the trap. A value of 0 means disarming will always work
+         * (e.g. for funnels), a values of 99 means it can not be disarmed at all. Smaller values
+         * makes it easier to disarm the trap.
+         */
         int get_difficulty() const
         {
             return difficulty;
         }
-        // Type of trap
+        /**
+         * If true, this is not really a trap and there won't be any safety queries before stepping
+         * onto it (e.g. for funnels).
+         */
         bool is_benign() const
         {
             return benign;
         }
-        // non-generic numbers for special cases
-        int funnel_radius_mm;
-        /** If an item with this weight or more is thrown onto the trap, it triggers. */
-        int trigger_weight;
         /** Player has not yet seen the trap and returns the variable chance, at this moment,
          of whether the trap is seen or not. */
         bool detect_trap( const tripoint &pos, const player &p ) const;
-        /** Can player/npc p see this kind of trap given their memory? */
+        /**
+         * Can player/npc p see this kind of trap, either by their memory (they known there is
+         * the trap) or by the visibility of the trap (the trap is not hidden at all)?
+         */
         bool can_see( const tripoint &pos, const player &p ) const;
-        /** Trigger trap effects by creature that stepped onto it. */
+        /**
+         * Trigger trap effects.
+         * @param creature The creature that triggered the trap, it does not necessarily have to
+         * be on the place of the trap (traps can be triggered from adjacent, e.g. when disarming
+         * them). This can also be a null pointer if the trap has been triggered by some thrown
+         * item (which must have the @ref trigger_weight).
+         * @param pos The location of the trap in the main map.
+         */
         void trigger( const tripoint &pos, Creature *creature ) const;
-
+        /**
+         * If the given item is throw onto the trap, does it trigger the trap?
+         */
+        bool triggered_by_item( const item &itm ) const;
+        /**
+         * Called when a trap at the given point in the main map has been disarmed.
+         * It should spawn trap items (if any) and remove the trap from the map via
+         * @ref map::remove_trap.
+         */
+        void on_disarmed( const tripoint &pos ) const;
+        /**
+         * Whether this kind of trap actually occupies a 3x3 area. Currently only blade traps
+         * do so.
+         */
+        bool is_3x3_trap() const;
```

Comment:
> there are a number of useful items that could be modded in using those.

Do you have examples because I lack any kind of creativity and it might help deciding how and what to import from json.